### PR TITLE
Feat(eos_designs): Support Pure L2 Spine in L2LS

### PR DESCRIPTION
**Pull request by mthiel117**
## Change Summary

Add support for a Pure L2 Spine within an L2LS design.

## Related Issue(s)

Fixes aristanetworks/avd-internal#46
Fixes aristanetworks/avd-internal#47
Fixes aristanetworks/avd-internal#48

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Separate L2 and L3 Spines into 2 node-type-keys like below.  The `l3spine` key will support SVI's and running a routing protocol.  The `spine` key will only support pure L2 functionality.  This feature fixes having to supply unnecessary variables for a pure L2 spine such as loopback_ipv4_pool, mlag_peer_l3_ipv4_pool and virtual_router_mac_address.

```yaml
  "l2ls":
    l3spine:
      type: l3spine
      connected_endpoints: true
      mlag_support: true
      network_services:
        l2: true
        l3: true
      default_overlay_routing_protocol: none
      default_underlay_routing_protocol: none
    spine:
      type: spine
      connected_endpoints: true
      mlag_support: true
      network_services:
        l2: true
      underlay_router: false
      uplink_type: port-channel
```

## How to test
molecule converge -s eos_designs-l2ls

Added molecule test scenarios (eos_designs-l2ls) for L2ONLY_SPINES and L2ONLY_LEAFS.

Generated configs were also validated on a DUT.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Comment by tgodaA** [comment]
I think we need document for l2ls like we added for MPLS, https://avd.sh/en/devel/roles/eos_designs/doc/fabric-topology-mpls-BETA.html

We can add it in a separate PR.

**Review by tgodaA — DISMISSED** [review]
LGTM

**Comment by mthiel117** [comment]
Yes we are planning a separate PR for documentation.

File: ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/L2ONLY-LEAF1.cfg [hunk 1]
```diff
@@ -0,0 +1,62 @@
+!RANCID-CONTENT-TYPE: arista
+!
+vlan internal order ascending range 1006 1199
+!
+transceiver qsfp default-mode 4x10G
+!
+service routing protocols model multi-agent
```
Comment (tgodaA) [inline ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/L2ONLY-LEAF1.cfg:7]: Do we need this for L2LS?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/L2ONLY-LEAF1.cfg:7]: nope - and same with ip routing

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/L2ONLY-LEAF1.cfg:7]: But both things have been like this for years, so we cannot change it in this PR. It should be fixed in AVD 4.0

Comment (tgodaA) [reply ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/L2ONLY-LEAF1.cfg:7]: Sure, I will open a fix for todo in 4.0

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/L2ONLY-LEAF1.cfg:7]: https://github.com/aristanetworks/ansible-avd/issues/1988

**Comment by ClausHolbechArista** [comment]
> I think we need document for l2ls like we added for MPLS, https://avd.sh/en/devel/roles/eos_designs/doc/fabric-topology-mpls-BETA.html
> 
> We can add it in a separate PR.

Yes. We will sort out documentation seperately. Most of the docs are the same, so we will focus on an example first with associated guide, and then add the details for l2ls to some of the existing pages.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Issue with ip routing on l2 switches is not new, so I have submitted separate issue.

**Pull request merged**